lib/Dialect/FIRRTL/Transforms/LowerIntrinsics.cpp

```diff
@@ -518,6 +518,132 @@ class CirctProbeConverter : public IntrinsicConverter {
 
 } // namespace
 
+// Replace range of values with new wires and return them.
+template <typename R>
+static SmallVector<Value> replaceResults(OpBuilder &b, R &&range) {
+  return llvm::map_to_vector(range, [&b](auto v) {
+    auto w = b.create<WireOp>(v.getLoc(), v.getType()).getResult();
+    v.replaceAllUsesWith(w);
+    return w;
+  });
+}
+
+// Check ports are all inputs, emit diagnostic if not.
+static ParseResult allInputs(ArrayRef<PortInfo> ports) {
```

Comment:
nit: LogicalResult is better for this?

Reply:
`ParseResult` is used for the helpers used in `check` so inclined to use it here as well -- `LogicalResult` can't be used in same way re:"parsing things check" code (`return parseFirst() || parseSecond() || ...;`).

So keeping for consistency there.

Reply:
(But thank you for the suggestion, LMK if you disagree)